FirebaseStorageSwift/Tests/Integration/StorageIntegration.swift

```diff
@@ -0,0 +1,620 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import FirebaseStorage
+import FirebaseStorageSwift
+import XCTest
+
+class StorageIntegration: XCTestCase {
+  var app: FirebaseApp!
+  var storage: Storage!
+  static var once = false
+
+  override class func setUp() {
+    FirebaseApp.configure()
+  }
+
+  override func setUp() {
+    super.setUp()
+    app = FirebaseApp.app()
+    storage = Storage.storage(app: app!)
+
+    if !StorageIntegration.once {
+      StorageIntegration.once = true
+      let setupExpectation = expectation(description: "setUp")
+
+      let largeFiles = ["ios/public/1mb"]
+      let emptyFiles =
+        ["ios/public/empty", "ios/public/list/a", "ios/public/list/b", "ios/public/list/prefix/c"]
+      setupExpectation.expectedFulfillmentCount = largeFiles.count + emptyFiles.count
+
+      do {
+        let bundle = Bundle(for: StorageIntegration.self)
+        let filePath = try XCTUnwrap(bundle.path(forResource: "1mb", ofType: "dat"),
+                                     "Failed to get filePath")
+        let data = try XCTUnwrap(try Data(contentsOf: URL(fileURLWithPath: filePath)),
+                                 "Failed to load file")
+
+        for largeFile in largeFiles {
+          let ref = storage.reference().child(largeFile)
+          _ = ref.putData(data) { result in
+            switch result {
+            case let .success(metadata):
+              XCTAssertEqual(metadata.name, "1mb")
+            case let .failure(error):
+              XCTFail("Unexpected error \(error) from putData")
+            }
+            setupExpectation.fulfill()
+          }
+        }
+        for emptyFile in emptyFiles {
+          let ref = storage.reference().child(emptyFile)
+          _ = ref.putData(data) { result in
+            switch result {
+            case let .success(metadata):
+              XCTAssertNotNil(metadata, "Metadata should not be nil")
+            case let .failure(error):
+              XCTFail("Unexpected error \(error) from putData")
+            }
+            setupExpectation.fulfill()
+          }
+        }
+        waitForExpectations()
+      } catch {
+        XCTFail("Error thrown setting up files in setUp")
+      }
+    }
+  }
+
+  override func tearDown() {
```

Comment:
Is this actually required? I suspect Swift will destroy the test case instance naturally.

Reply:
As of 2016, this was required to release resources since test cases were never deallocated. I'm not sure if this has changed since then. https://twitter.com/benjaminencz/status/758792216022126592

Reply:
Makes sense, though to actually release the associated app, don't you need to call `-[FIRApp deleteApp]`? Without that `app = nil` seems misleading at best.

Firestore keeps a map of instances around associated with each app/database pair. I can't find equivalent code in [`FIRStorageComponent`](https://github.com/firebase/firebase-ios-sdk/blob/b6cd99e30a9ad87d3f933bfef62ec4d7c0d78cd5/FirebaseStorage/Sources/FIRStorageComponent.m). So I suspect clearing `storage` may actually have an effect.

Reply:
Will leave this consistent with the other two integration test variations for now.